src/file-selector.ts

```diff
@@ -122,10 +122,16 @@ function flatten<T>(items: any[]): T[] {
 
 function fromDataTransferItem(item: DataTransferItem) {
     const file = item.getAsFile();
+
+    let fileAsEntry: FileSystemEntry | null = null;
+    if (typeof item.webkitGetAsEntry === 'function') {
+        fileAsEntry = item.webkitGetAsEntry();
+    }
+
```

Comment:
You can get rid of this if on line 111 you add this:
```
else if (entry) {
  return fromDataTransferItem(item, entry);
}
```

Because you already have the entry in the `toFilePromises` func.